__frzr-deploy

```diff
@@ -4,34 +4,101 @@ set -e
 set -o pipefail
 
 get_img_url() {
-    CHANNEL=$1
-
-    result=$(jq '[
-          sort_by(.created_at) |
-          reverse |
-          .[] |
-          { name: .name, prerelease: .prerelease, state: .assets[].state, url: .assets[].browser_download_url } |
-          select(.url|test("img")) |
-          select(.state=="uploaded")
-        ]')
-
-    if [ "$CHANNEL" == "testing" ]; then
-        result=$(echo $result | jq '[ .[] | select(.name|test("UNSTABLE")|not) ]')
-    elif [ "$CHANNEL" == "stable" ]; then
-        result=$(echo $result | jq '[ .[] | select(.prerelease==false) ]')
-    elif [ "$CHANNEL" != "unstable" ]; then
-        result=$(echo $result | jq "[ .[] | select(.url|contains(\"-${CHANNEL}_\")) ]")
-    fi
-
-    echo $result | jq 'first | .url' | sed 's/"//g'
+	CHANNEL=$1
+
+	# Order by creation date in reverse
+	result=$(jq 'sort_by(.published_at) | reverse')
+
+	# Always check for stable date
+	stable_release_date=$(echo "${result}" | jq -r '[ .[] |
+		select(.prerelease==false) ] |
+		first |
+		.published_at'
+	)
+	# Check for stable url, this is the latest that have prerelease == false
+	stable_download_url=$(echo "${result}" | jq -r '[ .[] |
+		select(.prerelease==false) ] |
+		first |
+		.assets[] |
+		select(.browser_download_url | test("img")) |
+		.browser_download_url'
+	)
+
+
+	# Filter channels by release naming conventions
+	if [[ "$CHANNEL" =~ ^[0-9]+\-?[0-9]*$ ]] ; then
+		# Check first for explicit version numbers between stable releases
+		# Useful for downgrading
+		result=$(echo "${result}" | jq -r "[ .[] |
+					select(.prerelease==false) |
+					select(.name|test(\" ${CHANNEL}\$\")) ] |
+					first |
+					.assets[] |
+					select(.browser_download_url | test(\"img\")) |
+					.browser_download_url"
+				)
+	elif [ "$CHANNEL" == "stable" ]; then
+		result=$stable_download_url
+	elif [ "$CHANNEL" == "testing" ]; then
+		# Testing channel have prerelease = true and no other tags
```

Comment:
We could just require testing to have a [testing] tag to simplify things. I think we discussed doing this in the past.